### DISCUSSION TITLE
Change primary color based on selected theme?

### DISCUSSION THREAD

**Discussion by Arcade1080** [mantinedev/mantine#5460] [Q&A] [answered]
Hello! 

Can we set the primary color on a theme basis? 

`
{
....
 primaryColor: {
light: "red"
dark: "green"
}
...
`

Thank yoouuuuu!

**Comment by AxelReid** [comment]
I'm trying to give black and white design for all components by default(for example button with black bg on light mode and white bg on dark)
Also it would be better to add `light`, `dark` props to all style props

**Comment by Toshinaki** [comment, marked as answer]
You can get `colorScheme` and change `primaryColor` based on it when passing your theme object to `MantineProvider`:
```tsx

import { useColorScheme } from '@mantine/hooks';
...

const colorScheme = useColorScheme();

return (
  <MantineProvider
    ...
    theme={{
      primaryColor: colorScheme  === 'light' ? 'red' : 'green'
    }}
    ...
  >
    ...
  </MantineProvider>
```
